lib/codeowners/cli/list.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Codeowners
+  module Cli
+    # List files owned by the given owner
+    class List < Base
+      option :from, default: 'origin/master'
+      option :to, default: 'HEAD'
+      option :verbose, default: false, type: :boolean, aliases: '-v'
+      desc 'by <owner>', <<~DESC
+        Lists all files owned by owner.
+        If no owner is specified, default owner is taken from the config file.
+      DESC
+      # option :local, default: false, type: :boolean, aliases: '-l'
+      # option :branch, default: '', aliases: '-b'
+      def by(owner = config.default_owner)
+        return if owner.empty?
+
+        Reporter.print "Checking ownership for '#{owner}'"
+        results = checker.list_files_for_owner(owner)
+        if results
+          print_results(results)
+        else
+          Reporter.print "Owner #{owner} not defined in .github/CODEOWNERS"
+        end
+      end
+
+      def initialize(args = [], options = {}, config = {})
+        super
+        @repo_base_path = `git rev-parse --show-toplevel`
+        if !@repo_base_path || @repo_base_path.empty?
+          raise 'You must be positioned in a git repository to use this tool'
+        end
+
+        @repo_base_path.chomp!
+        Dir.chdir(@repo_base_path)
+
+        @checker ||= config[:checker] || default_checker
+      end
+
+      default_task :by
+
+      private
+
+      attr_reader :checker
+
+      def default_checker
+        Codeowners::Checker.new(@repo_base_path, options[:from], options[:to])
+      end
+
+      def print_results(results)
+        results.each do |pattern, files|
+          Reporter.print "Pattern: '#{pattern}' - matches #{files.length} file(s)"
```

Comment:
Testing manually this message is useful when you have multiple matches, but if you have like a single file it seems too verbose.

```
Pattern: 'app/one/file.rb' - matches 1 file(s)
```

This seems not a pattern but a file. Maybe we can also make it easy to read removing repeated `Pattern:` wording 🤔 

Probably, also worth sorting results as we do not refer to specific lines from the file.